tests/testQuestionParams.js

```diff
@@ -0,0 +1,336 @@
+const ERR = require('async-stacktrace');
+const _ = require('lodash');
+const assert = require('chai').assert;
+const request = require('request');
+const cheerio = require('cheerio');
+
+const config = require('../lib/config');
+const sqldb = require('../prairielib/lib/sql-db');
+const sqlLoader = require('../prairielib/lib/sql-loader');
+const sql = sqlLoader.loadSqlEquiv(__filename);
+
+const helperServer = require('./helperServer');
+
+const locals = {};
+
+locals.siteUrl = 'http://localhost:' + config.serverPort;
+locals.baseUrl = locals.siteUrl + '/pl';
+locals.courseBaseUrl = locals.baseUrl + '/course/1';
+locals.courseInstanceBaseUrl = locals.baseUrl + '/course_instance/1';
+locals.courseInstanceBaseUrl = locals.baseUrl + '/course_instance/1';
+locals.instructorBaseUrl = locals.courseInstanceBaseUrl + '/instructor';
+locals.instructorAssessmentsUrl = locals.instructorBaseUrl + '/instance_admin/assessments';
+locals.instructorGradebookUrl = locals.instructorBaseUrl + '/instance_admin/gradebook';
+locals.questionBaseUrl = locals.courseInstanceBaseUrl + '/instance_question';
+locals.assessmentsUrl = locals.courseInstanceBaseUrl + '/assessments';
+locals.isStudentPage = false;
+
+const addNumbers1 = { qid: 'addNumbersParameterized/1', type: 'Freeform' };
+const addNumbers1_assessment = { qid: 'addNumbersParameterized/1', type: 'Freeform' }; // this one comes from an assessment
+const addNumbers2 = { qid: 'addNumbersParameterized/2', type: 'Freeform' };
+const addNumbers3 = { qid: 'addNumbersParameterized/3', type: 'Freeform' };
+const addNumbers4 = { qid: 'addNumbersParameterized/4', type: 'Freeform' };
+
+const questionsArray = [addNumbers1_assessment, addNumbers2, addNumbers3, addNumbers4];
+
+describe('Parameterized questions', function () {
+  this.timeout(60000);
+  var page;
+
+  before('set up testing server', helperServer.before());
+  after('shut down testing server', helperServer.after);
+
+  var elemList, res;
+
+  describe('the database', function () {
+    it('should contain questions', function (callback) {
+      sqldb.query(sql.select_questions, [], function (err, result) {
+        if (ERR(err, callback)) return;
+        if (result.rowCount === 0) {
+          return callback(new Error('no questions in DB'));
+        }
+        locals.questions = result.rows;
+        callback(null);
+      });
+    });
```

Comment:
You should use `async`/`await` here instead of callbacks. Ditto for the rest of the tests.